### PR TITLE
[TRTLLMINF-175][infra] Harden OpenSearchDB query/post with retry, exception safety, and structured logging

### PR DESCRIPTION
**Pull request by ZhanruiSunCh**
## Summary                                                                                                                                                                              
  - `queryFromOpenSearchDB` was vulnerable to unhandled exceptions from `requests.get()` (network timeout, DNS failure, connection refused) that would propagate to callers and break the 
  CI pipeline
  - `postToOpenSearchDB` had similar gaps in error classification and logging                                                                                                             
  - Added structured logging with `[OpenSearchDB][ERROR/WARN/INFO]` prefix for external log aggregation                                                                                   
                                                                                                                                                                                          
  ## Changes                                                                                                                                                                              
                                                                                                                                                                                          
  ### queryFromOpenSearchDB
  - Wrap `requests.get()` in try-catch so network exceptions trigger retry instead of crashing the caller
  - Add `json.dumps` serialization protection (TypeError/ValueError)
  - Distinguish retryable (429, 5xx) vs non-retryable (400-414) HTTP status codes
  - Add linear backoff between retry attempts (2s, 4s, 6s, 8s, max 10s)
  - Function now guarantees it never raises exceptions (returns None on any failure)

  ### postToOpenSearchDB
  - Add try-catch around `_postOnce` for network exceptions
  - Add backoff sleep between retry attempts
  - Classify HTTP errors consistently with query path

  ### Structured logging
  - `[OpenSearchDB][ERROR]`: terminal failures (config, non-retryable HTTP, exhausted, serialize)
  - `[OpenSearchDB][WARN]`: intermediate retry attempts (transient HTTP, network)
  - `[OpenSearchDB][INFO]`: normal control flow (local test skip, non-200 success)
  - `_sanitize_log()` replaces newlines and caps at 1000 chars for single-line grep

  ## Test plan
  - [x] 31 unit tests added (`jenkins/scripts/test_open_search_db.py`), all passing
  - [x] All HTTP requests mocked — no real network calls
  - [x] Covers: pre-validation, HTTP success/failure, network exceptions, retry/backoff, log format, edge cases
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Dev Engineer Review

- `OpenSearchDB` now handles serialization, network, and HTTP failures without propagating exceptions.
- Query and POST operations retry transient failures with capped backoff and fail fast on non-retryable 4xx responses.
- Structured logs include sanitized URLs, operation details, status, and retry state.
- `queryFromOpenSearchDB()` no longer declares a return type. Callers should verify compatibility with the changed API contract.
- The supplied test file focuses on query behavior. POST-specific behavior is not evident in the inspected test section.
- CI failed for commit `cd081c8`. Investigation and fixes are required before rerunning CI.
- Review severity counts are unavailable.

## QA Engineer Review

No test changes.

## Per-File QA Perspective

- `jenkins/scripts/open_search_db.py`: Verify query and POST behavior for validation, success, serialization errors, network exceptions, retryable responses, non-retryable 4xx responses, backoff limits, local-test bypass, and sanitized logs. Verify callers handle the removed query return annotation.
- `jenkins/scripts/test_open_search_db.py`: Adds mocked pytest coverage for query validation, successful responses, HTTP and network retries, backoff, serialization failures, dependency errors, log sanitization, and `_log()` edge cases. The file is outside `tests/` and is not listed in the supplied CI or manual-QA test lists.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

<!--
Please explain the issue and the solution in short.
-->

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: jenkins/scripts/open_search_db.py [hunk 1]
```diff
@@ -94,9 +94,22 @@ class OpenSearchDB:
...
+    # Tag format after the level: (op=query|post, cat=<category>, db=<project>[, status=N][, attempt=M/N])
+    _LOG_TAG = "[OpenSearchDB]"
+
     def __init__(self) -> None:
         pass
 
+    @staticmethod
+    def _sanitize_log(msg: str, max_len: int = 1000) -> str:
+        """Collapse newlines and cap length for single-line log output."""
+        return msg.replace("\n", " | ").replace("\r", "")[:max_len]
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py:111]: Acknowledged. In this module `project` values are always module-level string constants (e.g. `JOB_PROJECT_NAME = 'swdl-trtllm-infra-ci-prod-job_info'`) and `url` is derived from `OPEN_SEARCH_DB_BASE_URL` set by Jenkins — neither is arbitrary user input, so the practical injection risk is negligible here. Will add `_db = _S(project)` / `_url = _S(url)` sanitization as a follow-up hardening pass to keep this PR focused on functional correctness.

File: jenkins/scripts/open_search_db.py [hunk 2]
```diff
@@ -232,32 +245,40 @@ def postToOpenSearchDB(json_data, project) -> bool:
...
+                f"data type check failed")
             return False
 
         json_data_dump = json.dumps(json_data)
 
         if DISABLE_OPEN_SEARCH_DB_FOR_LOCAL_TEST:
-            OpenSearchDB.logger.info(
-                f"OpenSearchDB is disabled for local test, skip posting to OpenSearchDB: {json_data_dump}"
-            )
+            OpenSearchDB.logger.info(f"{_T}[INFO](op=post, db={project}) "
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py]: Fixed in commit `3ee969c`. `DISABLE_OPEN_SEARCH_DB_FOR_LOCAL_TEST` is now the very first check (after `use_poc_db` is computed), returning `True` immediately without requiring URL or credentials to be set.

File: jenkins/scripts/open_search_db.py [hunk 3]
```diff
@@ -232,32 +245,40 @@ def postToOpenSearchDB(json_data, project) -> bool:
...
             return False
         if not OpenSearchDB.typeCheckForOpenSearchDB(json_data):
-            OpenSearchDB.logger.info(
-                f"OpenSearchDB type check failed! json_data:{json_data}")
+            OpenSearchDB.logger.error(
+                f"{_T}[ERROR](op=post, cat=type_check, db={project}) "
+                f"data type check failed")
             return False
 
         json_data_dump = json.dumps(json_data)
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py]: Fixed in commit `3ee969c`. `json.dumps` in `postToOpenSearchDB` is now wrapped in `try/except (TypeError, ValueError)`, matching the existing guard in `queryFromOpenSearchDB`. Returns `False` with a structured `cat=serialize` error log on failure.

File: jenkins/scripts/open_search_db.py [hunk 4]
```diff
@@ -266,71 +287,117 @@ def postToOpenSearchDB(json_data, project) -> bool:
...
                     return True
-                else:
-                    OpenSearchDB.logger.info(
-                        f"OpenSearchDB post failed, will retry, error:{status_code} {last_text}"
-                    )
+
+                last_error = _S(f"HTTP {status_code}: {last_text[:200]}")
+                OpenSearchDB.logger.warning(
+                    f"{_T}[WARN](op=post, cat=transient, status={status_code}, db={project}, attempt={attempt}/{DEFAULT_RETRY_COUNT}) "
+                    f"will retry")
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py]: Fixed in commit `3ee969c`. Added `_is_non_retryable(status_code)` helper using a range check (`400 <= status_code < 500 and status_code \!= 429`) instead of the incomplete fixed set. Both POST and query paths now use the same classifier. The 418 test was updated to assert `call_count == 1` (immediate failure) rather than full retry.

File: jenkins/scripts/open_search_db.py [hunk 5]
```diff
@@ -266,71 +287,117 @@ def postToOpenSearchDB(json_data, project) -> bool:
...
             except Exception as e:
-                OpenSearchDB.logger.info(
-                    f"OpenSearchDB post exception, attempt {attempt + 1} error: {e}"
-                )
-        OpenSearchDB.logger.info(
-            f"Fail to postToOpenSearchDB after {DEFAULT_RETRY_COUNT} tries: {url}, json: {json_data_dump}, last error: {last_text}"
+                last_error = _S(f"{type(e).__name__}: {e}")
+                OpenSearchDB.logger.warning(
+                    f"{_T}[WARN](op=post, cat=network, db={project}, attempt={attempt}/{DEFAULT_RETRY_COUNT}) "
+                    f"url: {url}, error: {last_error}")
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py]: Fixed in commit `3ee969c`. `queryFromOpenSearchDB` now returns `None` immediately (with an ERROR log) when `requests is None`, avoiding the full backoff loop on `AttributeError`. The `except Exception` in the retry loop was narrowed to `except requests.exceptions.RequestException` so programming bugs propagate normally. Note: `postToOpenSearchDB` keeps `except Exception` intentionally — it delegates to `_postOnce` which can use either `requests` or `urllib`, so catching `RequestException` alone would miss `urllib.error.URLError` and similar.

File: jenkins/scripts/open_search_db.py [hunk 6]
```diff
@@ -266,71 +287,117 @@ def postToOpenSearchDB(json_data, project) -> bool:
...
     @staticmethod
     def queryFromOpenSearchDB(json_data, project) -> dict:
         """
         Query data from OpenSearchDB.
 
         :param json_data: Data to query, type dict or list.
         :param project: Name of the project.
-        :return: dict, query result.
+        :return: requests.Response on success, None on failure.
+                 This function never raises exceptions.
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py]: Fixed in commit `3ee969c`. Removed the incorrect `-> dict` annotation from `queryFromOpenSearchDB`. A proper `Optional[requests.Response]` annotation would require a `TYPE_CHECKING` guard due to the conditional `import requests` at module level; the docstring already documents the return contract, so removing the wrong annotation is the least-invasive fix.

File: jenkins/scripts/test_open_search_db.py [hunk 7]
```diff
@@ -0,0 +1,380 @@
...
+
+    def __init__(self, status_code, json_body=None, text=None):
+        self.status_code = status_code
+        self._json_body = json_body
+        self.text = (
+            text if text is not None else (json.dumps(json_body) if json_body is not None else "")
+        )
+
+    def json(self):
+        return self._json_body
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/test_open_search_db.py:45]: Noted. This is a CI/infra script, not a public library. The project's `CODING_GUIDELINES.md` type-annotation requirement targets production Python under `tensorrt_llm/`. Will add `-> None` and parameter annotations to test helpers in a follow-up cleanup commit to keep this PR's diff focused on functional fixes.

File: jenkins/scripts/test_open_search_db.py [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+        multiline_body = "<html>\n<body>\n<h1>503 Service Unavailable</h1>\n</body>\n</html>"
+        mock_get.return_value = MockResponse(400, text=multiline_body)
+        import logging
+
+        with caplog.at_level(logging.INFO):
+            OpenSearchDB.queryFromOpenSearchDB("{}", VALID_PROJECT)
+        for line in caplog.text.splitlines():
+            if f"{_TAG}[ERROR]" in line:
+                assert "\n" not in line
+                assert " | " in line
```
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/test_open_search_db.py]: Fixed in commit `3ee969c`. Replaced the `splitlines()` loop (where `assert '\n' not in line` was tautologically true) with a `caplog.records`-based check: collects all records whose `getMessage()` contains `[OpenSearchDB][ERROR]`, asserts at least one exists, then verifies each raw message contains no `\n` and does contain ` | `.

File: jenkins/scripts/open_search_db.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py]: Thanks for re-checking\! The push had gone to `NVIDIA/TensorRT-LLM` (origin) instead of the fork `ZhanruiSunCh/TensorRT-LLM`. Just pushed correctly — commit `3ee969c` is now on `user/zhanruis/0715_check_nvdf_read_and_write_api` in the fork. The fixes are: early `if requests is None: return None` before the retry loop, and `except requests.exceptions.RequestException` replacing the broad `except Exception`.

File: jenkins/scripts/open_search_db.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZhanruiSunCh) [reply jenkins/scripts/open_search_db.py:111]: No need to open a GitHub issue — I'll track the `_db = _S(project)` / `_url = _S(url)` hardening in the existing TRTLLMINF-175 Jira ticket as a follow-up item. Thanks for the offer\!

**Comment by ZhanruiSunCh** [comment]
@coderabbitai code review

**Comment by ZhanruiSunCh** [comment]
@coderabbitai Fixed in commit `cd081c8`. Both paths now guard the transient "will retry" warning with `attempt < DEFAULT_RETRY_COUNT`, so it is only emitted when there are remaining attempts.

- POST path: `if attempt < DEFAULT_RETRY_COUNT:` before the transient warning
- Query path: same guard applied

The network-exception warnings (which log the actual error type/message) are intentionally left unguarded — recording *what* failed on the final attempt is still useful for post-mortem diagnosis.
